File: joptimize/test/src/examples/narrow/Supertype2.scala
package joptimize.examples.narrow

object Supertype2 {
  def main(x: Int, y: Int): Int = {
    call(Bar, x) + call(Qux, y)
    // (x + 1) + (y + 2)
  }

  def call(f: Foo, n: Int): Int = call2(f, n)
  def call2(f: Foo, n: Int): Int = f.inc(n)
}

File: joptimize/test/src/examples/narrow/Supertype2.java
package joptimize.examples.narrow;

class Supertype2 {
  static int main(int x, int y){
    return call(new Bar(), x) + call(new Qux(), y);
    // (x + 1) + (y + 2)
  }

  static int call(Foo f, int n){ return call2(f, n); }
  static int call2(Foo f, int n){ return f.inc(n); }
}
